common/lib/api-client/middleware/request.js

```diff
@@ -13,8 +13,10 @@ function requestMiddleware({ cacheExpiry = 60, useRedisCache = false } = {}) {
       }
 
       const { req, jsonApi, cacheKey } = payload
+      const searchString = new URLSearchParams(req.params).toString()
```

Comment:
Does this want to use pickBy too?

Reply:
Interesting point actually. The request middleware is actually after the [`requestInclude`](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/2181f093f537546a9b04ce4a8dc1a05cb00a9aa1/common/lib/api-client/middleware/request-include.js#L22-L26) middleware that cleans up that key.

I wonder if we actually want to have something at the start of the middleware that cleans the params of any undefined values... @solidgoldpig what are your thoughts as I think you added the request include middleware?

Reply:
axios strips them out. Do it again here if you want ;)